### PR TITLE
Flux Geo package

### LINKED ISSUE #781: Document Flux experimental geo package
PR: https://github.com/influxdata/flux/pull/2565

### PR DESCRIPTION
**Pull request by sanderson**
Closes #781

Documents the new experimental Geo package.

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+  Import the `experimental/geo` package.
+menu:
+  v2_0_ref:
+    name: Geo
+    parent: Experimental
+weight: 201
+v2.0/tags: [functions, package, geo]
+---
+
+The Flux Geo package provides tools for working with geo-temporal data and performing
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: same commend as above. (if filtering and grouping geographic location isn't a subset of working w geo-temporal data, we could keep "and" in there:
...geo-temporal data and filtering and grouping geographic location.

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
+---
+title: Flux Geo package
+list_title: Geo package
+description: >
+  The Flux Geo package provides tools for working with geo-temporal data and performing
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: ...with geo-temporal data, such as filtering and grouping geographic locations.

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+##### Examples of geo-temporal line protocol
+```
+taxi,pt=start,s2_cell_id=89c2594 tip=3.75,dist=14.3,lat=40.744614,lon=-73.979424,tid=1572566401123234345i 1572566401947779410
+bike,id=biker-007,pt=via,s2_cell_id=89c25dc lat=40.753944,lon=-73.992035,tid=1572588100i 1572567115
+```
+
+## S2 Cell IDs
+Use **latitude** and **longitude** with the `s2.CellID.ToToken` endpoint of the S2
+Geometry Library to generate `s2_cell_id` tags.
+Specify your [S2 Cell ID level](https://s2geometry.io/resources/s2cell_statistics.html).
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: is level same as statistics?

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+{{% note %}}
+For faster filtering, use higher S2 Cell ID levels.
+But know that that higher levels increase
+[series cardinality](/v2.0/reference/glossary/#series-cardinality).
+{{% /note %}}
+
+Language-specific implementations of the S2 Geometry Library provide methods for
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: Is there a Go method we're using to generate tokens?

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+generating S2 Cell ID tokens.
+
+## Region definitions
+Many functions in the Geo package filter data based on geographic region.
+Define regions of different shapes using objects containing specific properties.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: Define geographic regions using the following shapes:
- box
- circle
- polygon

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+Regions can be the following shapes:
+
+### box
+Define a box-shaped region with an object containing the following properties:
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: ...region by specifying an object...

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+{
+  minLat: 40.51757813,
+  maxLat: 40.86914063,
+  minLon: -73.65234375,
+  maxLon: -72.94921875
+}
+```
+
+### circle
+Define a circular region with an object containing the following properties:
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: ...region by specifying an object...

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+
+- **lat**: latitude of the circle center in decimal degrees (WGS 84) _(Float)_
+- **lon**: longitude of the circle center in decimal degrees (WGS 84) _(Float)_
+- **radius**:  radius of the circle in Kilometers (km) _(Float)_
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: kilometers

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+Define a custom polygon region using an object containing the following properties:
+
+- **points**: points that define the custom polygon _(Array of objects)_
+
+    Define each point with an object containing the following properties:
+
+      - **lat**: latitude in decimal degrees (WGS 84) _(Float)_
+      - **lon**: longitude in decimal degrees (WGS 84) _(Float)_
+
+##### Example polygon region
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: polygonal

File: content/v2.0/reference/flux/stdlib/experimental/geo/astracks.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+_**Data type:** Array of strings_
+
+### orderBy
+Column to order results by.
+Default is `["_time"]`
+
+_**Data type:** Array of strings_
+
+## Examples
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/astracks.md]: Tracking a box-shaped region?

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 11]
```diff
@@ -0,0 +1,176 @@
+---
+title: geo.filterRows() function
+description: >
+  The geo.filterRows() function filters data by a specified geographic region with
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: is there non-strict filtering?

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+  maxSize: -1,
+  level: -1,
+  s2cellIDLevel: -1,
+  correlationKey: ["_time"],
+  strict: true
+)
+```
+
+## Parameters
+
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: ### region

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 13]
```diff
@@ -0,0 +1,176 @@
...
+The region containing the desired data points.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: the specified data points

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 14]
```diff
@@ -0,0 +1,176 @@
...
+_**Data type:** Integer_
+
+### maxSize
+Maximum number of tiles that cover the specified region.
+Default is `-1`.
+
+_**Data type:** Integer_
+
+### level
+Desired [cell level](https://s2geometry.io/resources/s2cell_statistics.html) of grid tiles.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: Specified cell level

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 15]
```diff
@@ -0,0 +1,176 @@
...
+_**Data type:** Object_
+
+### minSize
+Minimum number of tiles that cover the specified region.
+Default is `24`.
+
+_**Data type:** Integer_
+
+### maxSize
+Maximum number of tiles that cover the specified region.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: is tile a predefined standard?

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+Default is `-1`.
+
+_**Data type:** Integer_
+
+### level
+Desired [cell level](https://s2geometry.io/resources/s2cell_statistics.html) of grid tiles.
+Default is `-1`.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: didn't see -1 on the stats page...is that intentional?

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 17]
```diff
@@ -0,0 +1,176 @@
...
+{{% /warn %}}
+
+### s2cellIDLevel
+[S2 Cell level](https://s2geometry.io/resources/s2cell_statistics.html) in `s2_cell_id` tag.
+Default is `-1`.
+
+_**Data type:** Integer_
+
+{{% note %}}
+When set to `-1`, `geo.filterRows()` attempts to automatically detect the S2 Cell ID level.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: automatically detect outside of the level specified in the Level field?

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 18]
```diff
@@ -0,0 +1,176 @@
...
+{{% /note %}}
+
+### correlationKey
+List of columns used to uniquely identify a row for output.
+Default is `["_time"]`.
+
+_**Data type:** Array of strings_
+
+### strict
+Enable strict geographic data filtering.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: curious about strict

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 19]
```diff
@@ -0,0 +1,176 @@
...
+    region: {
+      minLat: 40.51757813,
+      maxLat: 40.86914063,
+      minLon: -73.65234375,
+      maxLon: -72.94921875
+    }
+  )
+```
+
+##### Approximately filter data in a circular region
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: diff between approximate and strict

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md [hunk 20]
```diff
@@ -0,0 +1,176 @@
...
+    region: {
+      lat: 40.69335938,
+      lon: -73.30078125,
+      radius: 20.0
+    }
+    strict: false
+  )
+```
+
+##### Filter data in a custom polygon region
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: what makes it custom?

File: content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md [hunk 21]
```diff
@@ -0,0 +1,133 @@
...
+  - /v2.0/reference/flux/stdlib/experimental/geo/strictfilter/
+  - /v2.0/reference/flux/stdlib/experimental/geo/filterRows/
+---
+
+The `geo.gridFilter()` function filters data by a specified geographic region.
+It compares input data to a set of S2 Cell ID tokens located in the specified [region](#region).
+
+{{% note %}}
+S2 Grid cells may not perfectly align with the defined region, so results may include
+data with coordinates outside the region, but inside S2 grid cells that are partially covered by the region.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md]: w coordinates that fall both inside and outside the specified region?

File: content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md [hunk 22]
```diff
@@ -0,0 +1,133 @@
...
+  maxSize: -1,
+  level: -1,
+  s2cellIDLevel: -1
+)
+```
+
+## Parameters
+
+### region
+The region containing the desired data points.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md]: Specify object properties for the shape?

File: content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md [hunk 23]
```diff
@@ -0,0 +1,133 @@
...
+_**Data type:** Integer_
+
+### maxSize
+Maximum number of tiles that cover the specified region.
+Default is `-1`.
+
+_**Data type:** Integer_
+
+### level
+Desired [cell level](https://s2geometry.io/resources/s2cell_statistics.html) of grid tiles.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md]: Specify S2 cell level?

File: content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md [hunk 24]
```diff
@@ -0,0 +1,133 @@
...
+
+_**Data type:** Integer_
+
+{{% warn %}}
+`level` is mutually exclusive with `minSize` and `maxSize` and must be less than
+or equal to `s2cellIDLevel`.
+{{% /warn %}}
+
+### s2cellIDLevel
+[S2 Cell level](https://s2geometry.io/resources/s2cell_statistics.html) in `s2_cell_id` tag.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md]: Specify the...to include in...tag?

File: content/v2.0/reference/flux/stdlib/experimental/geo/_index.md (hunk 4 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/_index.md]: Adding some examples here. This is going to be replaced once Telegraf 1.14 ships. Users will be able to use a new Telegraf processor to generate cell tokens from lat/lon coordinates.

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md (hunk 11 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: Yes, there is.

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md (hunk 12 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: Thank you! Nice catch!

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md (hunk 15 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: Changed "tiles" to "cells"

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md (hunk 16 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: Yes, -1 means the level isn't specified.

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md (hunk 17 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: `level` and `s2cellIDLevel` are different. `level` specifies what level to use when creating grid cells. `s2cellIDLevel` specifies what cell level **was** used to generate the cell ID token. It relates to the nature of the data coming in.

File: content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md (hunk 18 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/filterrows.md]: Added more detail

**Pull request merged**